src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -180,58 +182,77 @@ public int Seek(Vector<byte> byte0Vector)
 
             var block = _block;
             var index = _index;
-            var array = block.Array;
+            var following = block.End - index;
+            byte[] array;
+            var byte0 = byte0Vector[0];
+
             while (true)
             {
-                while (block.End == index)
+                while (following == 0)
                 {
-                    if (block.Next == null)
+                    var newBlock = block.Next;
+                    if (newBlock == null)
                     {
                         _block = block;
                         _index = index;
                         return -1;
                     }
-                    block = block.Next;
-                    index = block.Start;
-                    array = block.Array;
+                    index = newBlock.Start;
+                    following = newBlock.End - index;
+                    block = newBlock;
                 }
-                while (block.End != index)
+                array = block.Array;
+                while (following > 0)
                 {
-                    var following = block.End - index;
-                    if (following >= Vector<byte>.Count)
+#if !DEBUG 
+// Need unit tests to test Vector path
+                    // Check will be Jitted away https://github.com/dotnet/coreclr/issues/1079
+                    if (Vector.IsHardwareAccelerated)
                     {
-                        var data = new Vector<byte>(array, index);
-                        var byte0Equals = Vector.Equals(data, byte0Vector);
-
-                        if (byte0Equals.Equals(Vector<byte>.Zero))
+#endif
+                        if (following >= _vectorSpan)
```

Comment:
But as you say x64 coreclr

>    213:                         if (following >= _vectorSpan)
00007FFEA7EFD24F  mov         rcx,7FFEA7DF0B28h 
00007FFEA7EFD259  mov         edx,21h 
00007FFEA7EFD25E  call        00007FFF079EBC50 
00007FFEA7EFD263  mov         rdx,7FFEA7DF0BFCh 
00007FFEA7EFD26D  cmp         r14d,dword ptr [rdx] 
00007FFEA7EFD270  jl          00007FFEA7EFD316 
00007FFEA7EFD276  mov         edx,dword ptr [r13+8] 
00007FFEA7EFD27A  cmp         ebp,edx 
00007FFEA7EFD27C  jae         00007FFEA7EFD388 
00007FFEA7EFD282  lea         ecx,[rbp+0Fh] 
00007FFEA7EFD285  cmp         ecx,edx 
00007FFEA7EFD287  jae         00007FFEA7EFD388 
00007FFEA7EFD28D  movups      xmm0,xmmword ptr [r13+rbp+10h] 
   214:                         {
   215:                             var byte0Equals = Vector.Equals(new Vector<byte>(array, index), byte0Vector);
00007FFEA7EFD293  movups      xmm1,xmmword ptr [rsi] 
00007FFEA7EFD296  pcmpeqb     xmm0,xmm1 
00007FFEA7EFD29A  movaps      xmmword ptr [rsp+30h],xmm0 
   216: 
   217:                             if (byte0Equals.Equals(Vector<byte>.Zero))
00007FFEA7EFD29F  movaps      xmm0,xmmword ptr [rsp+30h] 
00007FFEA7EFD2A4  pxor        xmm1,xmm1 
00007FFEA7EFD2A8  movaps      xmm2,xmm0 
00007FFEA7EFD2AB  pcmpeqd     xmm2,xmm1 
00007FFEA7EFD2AF  pshufd      xmm3,xmm2,4Eh 
00007FFEA7EFD2B4  andps       xmm2,xmm3 
00007FFEA7EFD2B7  pshufd      xmm3,xmm2,1 
00007FFEA7EFD2BC  pand        xmm2,xmm3 
00007FFEA7EFD2C0  movd        edx,xmm2 
00007FFEA7EFD2C4  cmp         edx,0FFFFFFFFh 
00007FFEA7EFD2CA  sete        dl 
00007FFEA7EFD2CD  movzx       edx,dl 
00007FFEA7EFD2D0  test        edx,edx 
00007FFEA7EFD2D2  je          00007FFEA7EFD2E8 
   218:                             {
   219:                                 following -= _vectorSpan;
00007FFEA7EFD2D4  mov         rdx,7FFEA7DF0BFCh 
00007FFEA7EFD2DE  sub         r14d,dword ptr [rdx] 
   220:                                 index += _vectorSpan;
00007FFEA7EFD2E1  add         ebp,dword ptr [rdx] 
00007FFEA7EFD2E3  jmp         00007FFEA7EFD37A